core/src/main/java/io/paradoxical/cassieq/auth/SignedRequestAuthFilter.java

```diff
@@ -0,0 +1,171 @@
+public class SignedRequestAuthFilter<P extends Principal> extends AuthFilter<AuthorizedRequestCredentials, P> {
+    private static final Logger logger = getLogger(SignedRequestAuthFilter.class);
+
+    private final String accountNamePathParameter;
+    private final String keyAuthPrefix;
+
+    public SignedRequestAuthFilter(String accountNamePathParameter, final String keyAuthPrefix) {
+        this.accountNamePathParameter = accountNamePathParameter;
+        this.keyAuthPrefix = keyAuthPrefix;
+    }
+
+    @Override
+    public void filter(final ContainerRequestContext requestContext) throws IOException {
+        try {
+
+            final MultivaluedMap<String, String> pathParameters = requestContext.getUriInfo().getPathParameters();
+            if (pathParameters.containsKey(accountNamePathParameter)) {
+
+                final AccountName accountName = AccountName.valueOf(pathParameters.getFirst(accountNamePathParameter));
+
+                final RequestParameters headerRequestParameters = parseHeaderRequestParameters(requestContext, accountName);
+                final SignedUrlParameters signedUrlParameters = parseSignedUrlParameters(requestContext, accountName);
+
+                final AuthorizedRequestCredentials credentials =
+                        AuthorizedRequestCredentials.builder()
+                                                    .requestParameters(MoreObjects.firstNonNull(headerRequestParameters, signedUrlParameters))
+                                                    .build();
+
+                if (setPrincipal(requestContext, credentials)) {
+                    return;
+                }
+            }
+            else {
+                return;
+            }
+        }
+        catch (IllegalArgumentException e) {
+            logger.warn("Error decoding credentials", e);
+        }
+
+        throw new WebApplicationException(unauthorizedHandler.buildResponse(prefix, realm));
+    }
+
+    private RequestParameters parseHeaderRequestParameters(final ContainerRequestContext requestContext, final AccountName accountName) {
+        final String header = requestContext.getHeaders().getFirst(HttpHeaders.AUTHORIZATION);
+
+        if (!Strings.isNullOrEmpty(header)) {
+            final Splitter splitter = Splitter.on(' ').limit(2).trimResults();
+            final List<String> components = splitter.splitToList(header);
+
+            if (components.size() == 2) {
+                final String authScheme = components.get(0);
+
+                if (keyAuthPrefix.equalsIgnoreCase(authScheme)) {
+                    return AccountKeyParameters.builder()
+                                               .accountName(accountName)
+                                               .authKey(components.get(1))
+                                               .build();
+                }
+
+                if (prefix.equalsIgnoreCase(authScheme)) {
+                    return SignedRequestParameters.builder()
```

Comment:
yes the prefix is a common thing for authorization headers. cassieq configured it to "Signed"
other common values are "Basic" for basic auth, or "Bearer" for token/oauth